nh_eobs_mobile/static/src/js/nhlib.js

```diff
@@ -828,23 +828,40 @@ NHMobileForm = (function(superClass) {
         task_list = triggered_tasks ? triggered_tasks : pos;
         return new window.NH.NHModal('submit_success', server_data.title, task_list, buttons, 0, body);
       } else if (server_data.status === 'success' && data.status === 4) {
-        btn = '<a href="' + self.urls['task_list']().url + '" data-action="confirm" data-target="cancel_success">' + 'Go to My Tasks</a>';
-        return new window.NH.NHModal('cancel_success', server_data.title, '<p>' + server_data.desc + '</p>', [btn], 0, self.form);
+        triggered_tasks = '';
+        buttons = ['<a href="' + self.urls['task_list']().url + '" data-action="confirm" data-target="cancel_success">' + 'Go to My Tasks</a>'];
+        if (data.related_tasks.length === 1) {
+          triggered_tasks = '<p>' + data.related_tasks[0].summary + '</p>';
+          rt_url = self.urls['single_task'](data.related_tasks[0].id).url;
+          buttons.push('<a href="' + rt_url + '">Confirm</a>');
+        } else if (data.related_tasks.length > 1) {
+          tasks = '';
+          ref1 = data.related_tasks;
+          for (j = 0, len1 = ref1.length; j < len1; j++) {
+            task = ref1[j];
+            st_url = self.urls['single_task'](task.id).url;
+            tasks += '<li><a href="' + st_url + '">' + task.summary + '</a></li>';
+          }
+          triggered_tasks = '<ul class="menu">' + tasks + '</ul>';
+        }
+        pos = '<p>' + server_data.desc + '</p>';
+        task_list = triggered_tasks ? triggered_tasks : pos;
+        return new window.NH.NHModal('cancel_success', server_data.title, task_list, buttons, 0, self.form);
       } else {
         action_buttons = (function() {
-          var j, len1, ref1, ref2, results;
-          ref1 = self.form.elements;
+          var k, len2, ref2, ref3, results;
+          ref2 = self.form.elements;
           results = [];
-          for (j = 0, len1 = ref1.length; j < len1; j++) {
-            element = ref1[j];
-            if ((ref2 = element.getAttribute('type')) === 'submit' || ref2 === 'reset') {
+          for (k = 0, len2 = ref2.length; k < len2; k++) {
+            element = ref2[k];
+            if ((ref3 = element.getAttribute('type')) === 'submit' || ref3 === 'reset') {
```

Comment:
Ah okay, didn't realise this was generated code actually, but thanks for the explanation.